To-do list/Controllers/TaskListViewController.swift

```diff
@@ -23,109 +23,73 @@ class TaskListViewController: UIViewController, TaskTableViewCellDelegate {
+            type: TaskDetailsViewController.self)
+        taskDetailsVC.task = Task(title: "",
+                                  description: "",
+                                  deadline: .now,
+                                  status: .todo,
+                                  id: tasks.count+1)
+        taskDetailsVC.delegate = self
+        self.navigationController?.pushViewController(taskDetailsVC, animated: true)
     }
     
-    @IBAction func addButton(_ sender: UIButton) {
-        let taskDetailsVC = self.storyboard?.instantiateViewController(
-            withIdentifier: "TaskDetailsViewController") as? TaskDetailsViewController
-        taskDetailsVC?.userIndex = userIndex
-        taskDetailsVC?.choosenTableViewCell = false
-        self.navigationController?.pushViewController(taskDetailsVC!, animated: true)
+    func didTapStatusButton(cell: TaskTableViewCell, didClickOnStatus task: Task) {
+        if let index = tasks.firstIndex(where: { $0.id == task.id}) {
+            tasks[index].status = tasks[index].status.nextState
+        }
+        tasksTableView.reloadData()
     }
     
-    func didTapStatusButton(cell: TaskTableViewCell) {
-        if let chosenIndex = tasksTableView.indexPath(for: cell) {
-            var currentStatus = tasks[chosenIndex.row].status
-            
-            
-            switch currentStatus {
-            case "in progress":
-                currentStatus = "done"
-                print("current status: \(currentStatus)")
-                tasks[chosenIndex.row].status = currentStatus
-                tasksTableView.reloadData()
-                users[userIndex].task?[chosenIndex.row].status = currentStatus
-                UserDefaultsManager.shared.setValueForUser(value: users)
-            case "done":
-                currentStatus = "to do"
-                print("current status: \(currentStatus)")
-                tasks[chosenIndex.row].status = currentStatus
-                tasksTableView.reloadData()
-                users[userIndex].task?[chosenIndex.row].status = currentStatus
-                UserDefaultsManager.shared.setValueForUser(value: users)
-            case "to do":
-                currentStatus = "in progress"
-                print("current status: \(currentStatus)")
-                tasks[chosenIndex.row].status = currentStatus
-                tasksTableView.reloadData()
-                users[userIndex].task?[chosenIndex.row].status = currentStatus
-                UserDefaultsManager.shared.setValueForUser(value: users)
-            default:
-                break
-            }
+    func taskDetails(_ controller: TaskDetailsViewController, didCreateUpdate task: Task) {
+        if let index = tasks.firstIndex(where: { $0.id == task.id}) {
+            tasks[index] = task
+        } else {
+            tasks.append(task)
         }
+        tasksTableView.reloadData()
+    }
+    
+    private func sortedTasks() {
```

Comment:
the function name should always be - verb in infinitive form. rename to "sortTasks()"

Reply:
done